File: chebtst/bm_ext_plot_laml_power.m
clc;clear;close all;



result00 = load('bm_ext_base_laml_l_00e-3.mat'); 
Pplist00  = result00.Pplist;
result01 = load('bm_ext_base_laml_l_10e-3.mat'); 
Pplist01  = result01.Pplist;
result02 = load('bm_ext_base_laml_l_20e-3.mat'); 
Pplist02  = result02.Pplist;
result03 = load('bm_ext_base_laml_l_30e-3.mat'); 
Pplist03  = result03.Pplist;
result04 = load('bm_ext_base_laml_l_40e-3.mat'); 
Pplist04  = result04.Pplist;
result05 = load('bm_ext_base_laml_l_50e-3.mat'); 
Pplist05  = result05.Pplist;
result06 = load('bm_ext_base_laml_l_60e-3.mat'); 
Pplist06  = result06.Pplist;
result07 = load('bm_ext_base_laml_l_70e-3.mat');
Pplist07  = result07.Pplist;
result08 = load('bm_ext_base_laml_l_80e-3.mat'); 
Pplist08  = result08.Pplist;
result09 = load('bm_ext_base_laml_l_90e-3.mat'); 
Pplist09  = result09.Pplist;
result10 = load('bm_ext_base_laml_l_100e-3.mat'); 
Pplist10  = result10.Pplist;

fr = result00.fr;
Rl = result00.Rl;
xib  = result00.xib;

%% Rl from 1 ohm to 10 Mohm, different laml
f = figure(1);
set(gca, 'linewidth', 1.1, 'fontsize', 12, 'fontname', 'times')
set(gcf, 'color', 'w')
colormap lines
subplot(231)
mesh(fr,Rl,abs(Pplist00), 'DisplayName','\lambda_l=0.0')
set(gca,'XScale','log');
set(gca,'YScale','log');
set(gca,'ZScale','log');
subplot(232)
mesh(fr,Rl,abs(Pplist02), 'DisplayName','\lambda_l=0.2')
set(gca,'XScale','log');
set(gca,'YScale','log');
set(gca,'ZScale','log');
subplot(233)
mesh(fr,Rl,abs(Pplist04), 'DisplayName','\lambda_l=0.4')
set(gca,'XScale','log');
set(gca,'YScale','log');
set(gca,'ZScale','log');
subplot(234)
mesh(fr,Rl,abs(Pplist06), 'DisplayName','\lambda_l=0.6')
set(gca,'XScale','log');
set(gca,'YScale','log');
set(gca,'ZScale','log');
subplot(235)
mesh(fr,Rl,abs(Pplist08), 'DisplayName','\lambda_l=0.8')
set(gca,'XScale','log');
set(gca,'YScale','log');
set(gca,'ZScale','log');
subplot(236)
mesh(fr,Rl,abs(Pplist10), 'DisplayName','\lambda_l=1.0')
set(gca,'XScale','log');
set(gca,'YScale','log');
set(gca,'ZScale','log');


% legend('show')

% ax = findobj(f,'Type','Axes');
% for i=1:length(ax)
%     ylabel(ax(i),{'Nice'})
%     title(ax(i),{'Very Nice'})
% end




% title('Output voltage amplitude at different frequency and load')
% print(gcf,'fig_output_voltage_vs_fr_Rl_laml_all','-dpdf');
% print(gcf,'fig_output_voltage_vs_fr_Rl_laml_all','-depsc');
% print(gcf,'fig_output_voltage_vs_fr_Rl_laml_all','-djpeg','-r300','-noui');      

% 
% 
% %% Rl  1 ohm, different laml
% figure(2)
% plot(fr(1,:),abs(Pplist00(1,:)),'r', 'DisplayName','\lambda_l=0.0'); hold on
% plot(fr(1,:),abs(Pplist01(1,:)),'g', 'DisplayName','\lambda_l=0.1')
% plot(fr(1,:),abs(Pplist02(1,:)),'b', 'DisplayName','\lambda_l=0.2')
% plot(fr(1,:),abs(Pplist03(1,:)),'k', 'DisplayName','\lambda_l=0.3')
% plot(fr(1,:),abs(Pplist04(1,:)),'c', 'DisplayName','\lambda_l=0.4')
% plot(fr(1,:),abs(Pplist05(1,:)),'r.-', 'DisplayName','\lambda_l=0.5')
% plot(fr(1,:),abs(Pplist06(1,:)),'g.-', 'DisplayName','\lambda_l=0.6')
% plot(fr(1,:),abs(Pplist07(1,:)),'b.-', 'DisplayName','\lambda_l=0.7')
% plot(fr(1,:),abs(Pplist08(1,:)),'k.-', 'DisplayName','\lambda_l=0.8')
% plot(fr(1,:),abs(Pplist09(1,:)),'c.-', 'DisplayName','\lambda_l=0.9')
% plot(fr(1,:),abs(Pplist10(1,:)),'y', 'DisplayName','\lambda_l=1.0')
% set(gca,'XScale','log');
% set(gca,'YScale','log');
% xlabel('base excitation frequency');
% ylabel('output voltage amplitude');
% set(gca, 'linewidth', 1.1, 'fontsize', 12, 'fontname', 'times')
% set(gcf, 'color', 'w')
% legend('show')
% title('external electrical load R\_l = 1.0 \Omega')
% 
% 
% %% Rl  10 ohm, different laml
% figure(3)
% plot(fr(1,:),abs(Pplist00(2,:)),'r', 'DisplayName','\lambda_l=0.0'); hold on
% plot(fr(1,:),abs(Pplist01(2,:)),'g', 'DisplayName','\lambda_l=0.1')
% plot(fr(1,:),abs(Pplist02(2,:)),'b', 'DisplayName','\lambda_l=0.2')
% plot(fr(1,:),abs(Pplist03(2,:)),'k', 'DisplayName','\lambda_l=0.3')
% plot(fr(1,:),abs(Pplist04(2,:)),'c', 'DisplayName','\lambda_l=0.4')
% plot(fr(1,:),abs(Pplist05(2,:)),'r.-', 'DisplayName','\lambda_l=0.5')
% plot(fr(1,:),abs(Pplist06(2,:)),'g.-', 'DisplayName','\lambda_l=0.6')
% plot(fr(1,:),abs(Pplist07(2,:)),'b.-', 'DisplayName','\lambda_l=0.7')
% plot(fr(1,:),abs(Pplist08(2,:)),'k.-', 'DisplayName','\lambda_l=0.8')
% plot(fr(1,:),abs(Pplist09(2,:)),'c.-', 'DisplayName','\lambda_l=0.9')
% plot(fr(1,:),abs(Pplist10(2,:)),'y', 'DisplayName','\lambda_l=1.0')
% set(gca,'XScale','log');
% set(gca,'YScale','log');
% xlabel('base excitation frequency');
% ylabel('output voltage amplitude');
% set(gca, 'linewidth', 1.1, 'fontsize', 12, 'fontname', 'times')
% set(gcf, 'color', 'w')
% legend('show')
% title('external electrical load R\_l = 10.0 \Omega')
% 
% 
% 
% %% Rl  100 ohm, different laml
% figure(4)
% plot(fr(1,:),abs(Pplist00(3,:)),'r', 'DisplayName','\lambda_l=0.0'); hold on
% plot(fr(1,:),abs(Pplist01(3,:)),'g', 'DisplayName','\lambda_l=0.1')
% plot(fr(1,:),abs(Pplist02(3,:)),'b', 'DisplayName','\lambda_l=0.2')
% plot(fr(1,:),abs(Pplist03(3,:)),'k', 'DisplayName','\lambda_l=0.3')
% plot(fr(1,:),abs(Pplist04(3,:)),'c', 'DisplayName','\lambda_l=0.4')
% plot(fr(1,:),abs(Pplist05(3,:)),'r.-', 'DisplayName','\lambda_l=0.5')
% plot(fr(1,:),abs(Pplist06(3,:)),'g.-', 'DisplayName','\lambda_l=0.6')
% plot(fr(1,:),abs(Pplist07(3,:)),'b.-', 'DisplayName','\lambda_l=0.7')
% plot(fr(1,:),abs(Pplist08(3,:)),'k.-', 'DisplayName','\lambda_l=0.8')
% plot(fr(1,:),abs(Pplist09(3,:)),'c.-', 'DisplayName','\lambda_l=0.9')
% plot(fr(1,:),abs(Pplist10(3,:)),'y', 'DisplayName','\lambda_l=1.0')
% % set(gca,'XScale','log');
% % set(gca,'YScale','log');
% xlim([1,100]);
% xlabel('base excitation frequency');
% ylabel('output voltage amplitude');
% set(gca, 'linewidth', 1.1, 'fontsize', 12, 'fontname', 'times')
% set(gcf, 'color', 'w')
% legend('show')
% title('external electrical load R\_l = 100.0 \Omega')




%% Rl  1000000 ohm, different laml
% figure(8)
% plot(fr(1,:),abs(Pplist00(7,:)),'r', 'DisplayName','\lambda_l=0.0'); hold on
% plot(fr(1,:),abs(Pplist01(7,:)),'g', 'DisplayName','\lambda_l=0.1')
% plot(fr(1,:),abs(Pplist02(7,:)),'b', 'DisplayName','\lambda_l=0.2')
% plot(fr(1,:),abs(Pplist03(7,:)),'k', 'DisplayName','\lambda_l=0.3')
% plot(fr(1,:),abs(Pplist04(7,:)),'c', 'DisplayName','\lambda_l=0.4')
% plot(fr(1,:),abs(Pplist05(7,:)),'r.-', 'DisplayName','\lambda_l=0.5')
% plot(fr(1,:),abs(Pplist06(7,:)),'g.-', 'DisplayName','\lambda_l=0.6')
% plot(fr(1,:),abs(Pplist07(7,:)),'b.-', 'DisplayName','\lambda_l=0.7')
% plot(fr(1,:),abs(Pplist08(7,:)),'k.-', 'DisplayName','\lambda_l=0.8')
% plot(fr(1,:),abs(Pplist09(7,:)),'c.-', 'DisplayName','\lambda_l=0.9')
% plot(fr(1,:),abs(Pplist10(7,:)),'y', 'DisplayName','\lambda_l=1.0')
% % set(gca,'XScale','log');
% set(gca,'YScale','log');
% xlim([1,100]);
% xlabel('base excitation frequency');
% ylabel('output voltage amplitude');
% set(gca, 'linewidth', 1.1, 'fontsize', 12, 'fontname', 'times')
% set(gcf, 'color', 'w')
% legend('show')
% title('external electrical load R\_l = 1000000.0 \Omega')

File: chebtst/bm_ext_plot_laml_power.py
from mpl_toolkits.mplot3d import Axes3D
import numpy as np
import matplotlib.pyplot as plt
import scipy.io as sio
from matplotlib import cm
from matplotlib.ticker import LinearLocator, FormatStrFormatter


result00 = sio.loadmat('bm_ext_base_laml_l_00e-3.mat'); 
vplist00  = result00['Vplist'];
result01 = sio.loadmat('bm_ext_base_laml_l_10e-3.mat'); 
vplist01  = result01['Vplist'];
result02 = sio.loadmat('bm_ext_base_laml_l_20e-3.mat'); 
vplist02  = result02['Vplist'];
result03 = sio.loadmat('bm_ext_base_laml_l_30e-3.mat'); 
vplist03  = result03['Vplist'];
result04 = sio.loadmat('bm_ext_base_laml_l_40e-3.mat'); 
vplist04  = result04['Vplist'];
result05 = sio.loadmat('bm_ext_base_laml_l_50e-3.mat'); 
vplist05  = result05['Vplist'];
result06 = sio.loadmat('bm_ext_base_laml_l_60e-3.mat'); 
vplist06  = result06['Vplist'];
result07 = sio.loadmat('bm_ext_base_laml_l_70e-3.mat');
vplist07  = result07['Vplist'];
result08 = sio.loadmat('bm_ext_base_laml_l_80e-3.mat'); 
vplist08  = result08['Vplist'];
result09 = sio.loadmat('bm_ext_base_laml_l_90e-3.mat'); 
vplist09  = result09['Vplist'];
result10 = sio.loadmat('bm_ext_base_laml_l_100e-3.mat'); 
vplist10  = result10['Vplist'];


fr        = result01['fr'];
Rl        = result01['Rl'];
xib       = result01['xib'];

fig2 = plt.figure(2, figsize=(12,8))
plt.plot(fr[0,:],np.abs(vplist00[5,:]),'r', label= '$\lambda_l$=0.0')
plt.plot(fr[0,:],np.abs(vplist01[5,:]),'g', label= '$\lambda_l$=0.1')
plt.plot(fr[0,:],np.abs(vplist02[5,:]),'b', label= '$\lambda_l$=0.2')
plt.plot(fr[0,:],np.abs(vplist03[5,:]),'k', label= '$\lambda_l$=0.3')
plt.plot(fr[0,:],np.abs(vplist04[5,:]),'c', label= '$\lambda_l$=0.4')
plt.plot(fr[0,:],np.abs(vplist05[5,:]),'r.--', label= '$\lambda_l$=0.5')
plt.plot(fr[0,:],np.abs(vplist06[5,:]),'g.--', label= '$\lambda_l$=0.6')
plt.plot(fr[0,:],np.abs(vplist07[5,:]),'b.--', label= '$\lambda_l$=0.7')
plt.plot(fr[0,:],np.abs(vplist08[5,:]),'k.--', label= '$\lambda_l$=0.8')
plt.plot(fr[0,:],np.abs(vplist09[5,:]),'c.--', label= '$\lambda_l$=0.9')
plt.plot(fr[0,:],np.abs(vplist10[5,:]),'y', label= '$\lambda_l$=1.0')
plt.xlabel('Base excitation frequency $fr$')
plt.ylabel('Amplitude of the output voltage $V_p$')
plt.title('Amplitude of the output voltage $V_p$ versus base excitation frequency $fr$ when $R_l = 1000\ \Omega$')
plt.legend(loc='lower right')
# plt.xscale('log')
plt.yscale('log')
plt.xlim(1,100)

# plt.savefig('fig_output_voltage_vs_frequency_laml.jpg',dpi=300)
# plt.savefig('fig_output_voltage_vs_frequency_laml.eps')
# plt.savefig('fig_output_voltage_vs_frequency_laml.pdf')



## plot 2



fig = plt.figure(1)
ax  = fig.gca(projection='3d')
# surf01 = ax.plot_surface(np.log10(fr), np.log10(Rl), np.log10(np.abs(vplist01)), cmap=cm.coolwarm,
#                        linewidth=0, antialiased=False)
# surf02 = ax.plot_surface(np.log10(fr), np.log10(Rl), np.log10(np.abs(vplist02)), cmap=cm.coolwarm,
#                        linewidth=0, antialiased=False)
# surf03 = ax.plot_surface(np.log10(fr), np.log10(Rl), np.log10(np.abs(vplist03)), cmap=cm.coolwarm,
#                        linewidth=0, antialiased=False)
# surf04 = ax.plot_surface(np.log10(fr), np.log10(Rl), np.log10(np.abs(vplist04)), cmap=cm.coolwarm,
#                        linewidth=0, antialiased=False)
# surf05 = ax.plot_surface(np.log10(fr), np.log10(Rl), np.log10(np.abs(vplist05)), cmap=cm.coolwarm,
#                        linewidth=0, antialiased=False)
# fr = np.log10(fr)
# Rl = np.log10(Rl)


surf01 = ax.plot_surface(fr, Rl, np.abs(vplist01), cmap=cm.coolwarm,
                       linewidth=0, antialiased=False)
# surf02 = ax.plot_surface(fr, Rl, np.abs(vplist02), cmap=cm.coolwarm,
#                        linewidth=0, antialiased=False)
# surf03 = ax.plot_surface(fr, Rl, np.abs(vplist03), cmap=cm.coolwarm,
#                        linewidth=0, antialiased=False)
# surf04 = ax.plot_surface(fr, Rl, np.abs(vplist04), cmap=cm.coolwarm,
#                        linewidth=0, antialiased=False)
# surf05 = ax.plot_surface(fr, Rl, np.abs(vplist05), cmap=cm.coolwarm,
#                        linewidth=0, antialiased=False)
# ax.set_zlim(-10.0, 10.0)
# ax.xaxis.set_scale('log')
# ax.yaxis.set_scale('log')
# ax.zaxis.set_scale('log')
# ax.set_zscale('log')
# ax.set_xscale('log')
# print(dir(ax.xaxis))
ax.zaxis.set_major_locator(LinearLocator(10))
ax.zaxis.set_major_formatter(FormatStrFormatter('%.02f'))
# fig.colorbar(surf, shrink=0.5, aspect=5)


# xticks = [1e0,1e1,1e2,1e3]
# yticks = [1e0,1e1,1e2,1e3,1e4,1e5,1e6,1e7]
# zticks = np.logspace(-5,5,11)
# ax.set_xticks(np.log10(xticks))
# ax.set_xticklabels(xticks)
# ax.set_yticks(np.log10(yticks))
# ax.set_yticklabels(yticks)
# ax.set_zticks(np.log10(zticks))
# ax.set_zticklabels(zticks)


plt.show()



